src/foam/dao/PgConnectionPool.java

```diff
@@ -0,0 +1,32 @@
+package foam.dao;
+
+import org.apache.commons.dbcp2.BasicDataSource;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+public abstract class PgConnectionPool {
```

Comment:
Is this Postgres specific, or is this just generic JDBC? Would the same code work for MySQL or Oracle?